packages/fuels-accounts/src/provider/retryable_client.rs

```diff
@@ -306,6 +307,22 @@ impl RetryableClient {
         })
         .await
     }
+
+    pub async fn contract_exists(&self, contract_id: &ContractId) -> RequestResult<bool> {
+        self.wrap(|| {
+            let query = ContractExistsQuery::build(ContractByIdArgs {
+                id: (*contract_id).into(),
+            });
+            self.client.query(query)
+        })
+        .await
+        .map(|query| {
+            query
+                .contract
+                .map(|contract| ContractId::from(contract.id) == *contract_id)
+                .unwrap_or(false)
+        })
```

Comment:
This assertion looks awkward, but IMO is better than just `query.contract.is_some()`, and `cynic` (or GraphQL in general, I don't know) requires at least 1 field in the query nonetheless (the simplest for `Contract` being its `id`)